poplar.mk

```diff
@@ -0,0 +1,304 @@
+
+UBOOT_BIN		?= $(UBOOT_PATH)/u-boot.bin
+OPTEE_BIN		?= $(OPTEE_PATH)/out/arm/core/tee-header_v2.bin
+OPTEE_BIN_EXTRA1	?= $(OPTEE_PATH)/out/arm/core/tee-pager_v2.bin
+OPTEE_BIN_EXTRA2	?= $(OPTEE_PATH)/out/arm/core/tee-pageable_v2.bin
+
+ROOTFS_BIN		?= linaro-stretch-developer-20170914-80.tar.gz
+ROOTFS_URL		?= http://download.djjproject.com/poplar/$(ROOTFS_BIN)
+
+PKG_OPTEE_VERSION	?= 1.0.0
+PKG_PATH		?= $(OUT_PATH)/debpkg/optee_$(PKG_OPTEE_VERSION)
+PKG_USR_BIN		?= $(PKG_PATH)/usr/bin
+PKG_CONTROL		?= $(PKG_PATH)/DEBIAN/control
+
+################################################################################
+# Targets
+################################################################################
+.PHONY: all
+all: u-boot arm-tf l-loader linux rootfs prepare-images deb-package | toolchains
+
+.PHONY: clean
+clean: u-boot-clean arm-tf-clean l-loader-clean linux-clean optee-os-clean
+
+################################################################################
+# Toolchain
+################################################################################
+include toolchain.mk
+
+################################################################################
+# Folders
+################################################################################
+$(OUT_PATH):
+	@mkdir -p $@
+
+################################################################################
+# U-Boot
+################################################################################
+.PHONY: u-boot-config
+u-boot-config:
+ifeq ($(wildcard $(UBOOT_PATH)/.config),)
+	$(MAKE) -C $(UBOOT_PATH) \
+		CROSS_COMPILE=$(AARCH64_CROSS_COMPILE) poplar_defconfig
+endif
+
+.PHONY: u-boot-menuconfig
+u-boot-menuconfig: u-boot-config
+	$(MAKE) -C $(UBOOT_PATH) \
+		CROSS_COMPILE=$(AARCH64_CROSS_COMPILE) menuconfig
+
+.PHONY: u-boot
+u-boot: u-boot-config
+	$(MAKE) -C $(UBOOT_PATH) \
+		CROSS_COMPILE="$(AARCH64_CROSS_COMPILE)"
+
+.PHONY: u-boot-clean
+u-boot-clean:
+	cd $(UBOOT_PATH) && git clean -xdf
+
+################################################################################
+# OP-TEE client
+################################################################################
+.PHONY: optee-client
+optee-client:
+	$(MAKE) -C $(OPTEE_CLIENT_PATH) \
+		CROSS_COMPILE="$(AARCH64_CROSS_COMPILE)"
+
+.PHONY: optee-client-clean
+optee-client-clean:
+	$(MAKE) -C $(OPTEE_CLIENT_PATH) clean
+
+################################################################################
+# OP-TEE xtest
+################################################################################
+.PHONY: optee-test
+optee-test: optee-os optee-client
+	$(MAKE) -C $(OPTEE_TEST_PATH) \
+		CROSS_COMPILE_HOST="$(AARCH64_CROSS_COMPILE)" \
+		CROSS_COMPILE_TA="$(AARCH32_CROSS_COMPILE)" \
+		TA_DEV_KIT_DIR="$(OPTEE_PATH)/out/arm/export-ta_arm32" \
+		OPTEE_CLIENT_EXPORT="$(OPTEE_CLIENT_PATH)/out/export"
+
+.PHONY: optee-test-clean
+optee-test-clean:
+	$(MAKE) -C $(OPTEE_TEST_PATH) clean
+
```

Comment:
Can't you just depend on targets in `common.mk` for `optee-client`, `optee-client-clean`, `optee-test` and `optee-test-clean`?